python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source.py

```diff
@@ -106,58 +128,81 @@ class Source:
-                        replace=transform.get(constants.CROISSANT_REPLACE),
-                        separator=transform.get(constants.CROISSANT_SEPARATOR),
+                        format=transform.get(str(constants.ML_COMMONS_FORMAT)),
+                        regex=transform.get(str(constants.ML_COMMONS_REGEX)),
+                        replace=transform.get(str(constants.ML_COMMONS_REPLACE)),
+                        separator=transform.get(str(constants.ML_COMMONS_SEPARATOR)),
                     )
                     for transform in transforms
                 )
                 # Safely access and check "data_extraction" from JSON-LD.
-                data_extraction = json_ld.get(constants.CROISSANT_DATA_EXTRACTION, {})
+                data_extraction = jsonld.get(
+                    str(constants.ML_COMMONS_DATA_EXTRACTION), {}
+                )
                 if isinstance(data_extraction, list) and data_extraction:
                     data_extraction = data_extraction[0]
+                # Remove the JSON-LD @id property if it exists:
+                data_extraction.pop("@id", None)
                 if len(data_extraction) > 1:
                     issues.add_error(
                         f"{constants.ML_COMMONS_DATA_EXTRACTION} should have one of the"
                         f" following properties: {constants.ML_COMMONS_FORMAT},"
                         f" {constants.ML_COMMONS_REGEX},"
-                        f" {constants.CROISSANT_REPLACE} or"
+                        f" {constants.ML_COMMONS_REPLACE} or"
                         f" {constants.ML_COMMONS_SEPARATOR}"
                     )
                 # Safely access and check "uid" from JSON-LD.
-                distribution = json_ld.get(constants.CROISSANT_DISTRIBUTION)
-                field = json_ld.get(constants.CROISSANT_FIELD)
+                distribution = jsonld.get(str(constants.SCHEMA_ORG_DISTRIBUTION))
+                field = jsonld.get(str(constants.ML_COMMONS_FIELD))
                 if distribution is not None and field is None:
                     uid = distribution
+                    node_type = "distribution"
                 elif distribution is None and field is not None:
                     uid = field
+                    node_type = "field"
                 else:
                     uid = None
+                    node_type = None
                     issues.add_error(
                         f"Every {constants.ML_COMMONS_SOURCE} should declare either"
                         f" {constants.ML_COMMONS_FIELD} or"
                         f" {constants.SCHEMA_ORG_DISTRIBUTION}"
                     )
                 # Safely access and check "file_property" from JSON-LD.
-                file_property = data_extraction.get(constants.CROISSANT_FILE_PROPERTY)
+                file_property = data_extraction.get(
+                    str(constants.ML_COMMONS_FILE_PROPERTY)
```

Comment:
Why do you need to use ML_COMMONS instead of CROISSANT?

Reply:
Acknowledged.

We don't use `CROISSANT_` constants anymore. They will be deleted in a coming PR.